c-sharp-razor/src/Minitwit.Infrastructure/MinitwitDbContext.cs

```diff
@@ -0,0 +1,146 @@
+using System.Configuration;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Identity.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore;
+using Minitwit.Core.Entities;
+
+
+namespace Minitwit.Infrastructure;
+
+/// <summary>
+///  EF Core will use the properties of the entities to create and control
+///  the database, without having the application directly interact with the database.
+/// </summary>
+
+public sealed class MinitwitDbContext : IdentityDbContext<Author, IdentityRole<int>, int>
+{
+    public DbSet<Message> Messages { get; set; } = null!;
+
+    public DbSet<Follow> Follows { get; set; } = null!;
+
+    public DbSet<Latest> LatestEntries { get; set; } = null!;
+
+    public MinitwitDbContext(DbContextOptions<MinitwitDbContext> dbContextOptions)
+        : base(dbContextOptions)
+    {
+        ChangeTracker.LazyLoadingEnabled = false;
+    }
+
+    protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
+    {
+        if (!optionsBuilder.IsConfigured)
+        {
+            // Get PostgreSQL environment variables
+            string username = Environment.GetEnvironmentVariable("POSTGRES_USER") ?? "user";
+            string password = Environment.GetEnvironmentVariable("POSTGRES_PASSWORD") ?? "pass";
+            string host = "192.168.0.146"; // Environment.GetEnvironmentVariable("POSTGRES_HOST") ?? "192.168.0.146" 
+            string port = Environment.GetEnvironmentVariable("POSTGRES_PORT") ?? "5432";
+            string database = Environment.GetEnvironmentVariable("POSTGRES_DB") ?? "waect";
+
+            // Construct PostgreSQL connection string
+            var connectionString = $"Host={host};Port={port};Database={database};Username={username};Password={password}";
```

Comment:
Is it possible to use the DATABASE_URL env var?

Reply:
_agree with Mr. Meding_

Reply:
yes, changed it!